javascripts/discourse/components/featured-homepage-topics.js

```diff
@@ -1,83 +1,116 @@
+            let featuredTagTopics = [];
+
+            topicList.topics.forEach((topic) =>
+              featuredTagTopics.push(Topic.create(topic))
+            );
+
+            this.set(
+              "featuredTagTopics",
+              featuredTagTopics.slice(0, settings.number_of_topics)
+            );
+          });
+      }
+    }
+  },
+
+  _checkClass() {
+    if (!this.showHere) {
+      document.querySelector("body").classList.remove(FEATURED_CLASS);
+    }
+  },
+
   init() {
     this._super(...arguments);
-    let sortOrder = settings.sort_by_created ? "created" : "activity";
-    if (settings.featured_tag) {
-      this.store
-        .findFiltered("topicList", {
-          filter: "latest",
-          params: {
-            tags: [`${settings.featured_tag}`],
-            order: sortOrder,
-          },
-        })
-        .then((topicList) => {
-          let featuredTagTopics = [];
+    this._getBanner();
+    this._checkClass();
+  },
 
-          topicList.topics.forEach((topic) =>
-            topic.image_url ? featuredTagTopics.push(Topic.create(topic)) : ""
-          );
+  didInsertElement() {
+    this.appEvents.on("page:changed", this, "_checkClass");
+  },
 
-          this.set(
-            "featuredTagTopics",
-            featuredTagTopics.slice(0, settings.number_of_topics)
-          );
-        });
-    }
+  willDestroyElement() {
+    document.querySelector("body").classList.remove(FEATURED_CLASS);
   },
 
   @discourseComputed("router.currentRoute", "router.currentRouteName")
   showHere(currentRoute, currentRouteName) {
+    let showHere;
+
     if (currentRoute) {
       if (settings.show_on === "homepage") {
-        return currentRouteName === `discovery.${defaultHomepage()}`;
+        showHere = currentRouteName === `discovery.${defaultHomepage()}`;
       } else if (settings.show_on === "top_menu") {
         const topMenuRoutes = this.siteSettings.top_menu
           .split("|")
           .filter(Boolean);
-        return topMenuRoutes.includes(currentRoute.localName);
+        showHere = topMenuRoutes.includes(currentRoute.localName);
       } else if (settings.show_on === "all") {
-        return (
+        showHere =
           currentRouteName.indexOf("editCategory") &&
           currentRouteName.indexOf("admin") &&
-          currentRouteName.indexOf("full-page-search")
-        );
+          currentRouteName.indexOf("full-page-search");
       } else {
         return false;
       }
     }
+
+    if (showHere) {
+      document.querySelector("body").classList.add(FEATURED_CLASS);
+    }
+
+    return showHere;
   },
 
   @discourseComputed()
   showTitle() {
     if (settings.show_title) {
       const titleElement = document.createElement("h2");
-      titleElement.innerHTML = settings.title_text;
+      titleElement.innerHTML = I18n.t(themePrefix("featured_topic_title"))
+        .length
+        ? I18n.t(themePrefix("featured_topic_title"))
+        : settings.title_text;
```

Comment:
```suggestion
      titleElement.innerHTML = I18n.t(themePrefix("featured_topic_title")) || settings.title_text;
```

(+a prettier pass; I'm sure it will inject a newline somewhere)